### PR TITLE
PWGCF: upload PbPb flow analysis for hyperloop

### PR DESCRIPTION
**Pull request by Luzhiyongg**
the code to produce flow observables for PbPb run3 data

### CODE REVIEW

File: PWGCF/Flow/Tasks/FlowPbPbTask.cxx [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+    else if (SampleIndex == 5)
+      FillBootstrap<5>(cent, ptSum_Gap08, weffEvent_WithinGap08, sum_pt_wSquare_WithinGap08, sum_ptSquare_wSquare_WithinGap08, WeffEvent_diff_WithGap08);
+    else if (SampleIndex == 6)
+      FillBootstrap<6>(cent, ptSum_Gap08, weffEvent_WithinGap08, sum_pt_wSquare_WithinGap08, sum_ptSquare_wSquare_WithinGap08, WeffEvent_diff_WithGap08);
+    else if (SampleIndex == 7)
+      FillBootstrap<7>(cent, ptSum_Gap08, weffEvent_WithinGap08, sum_pt_wSquare_WithinGap08, sum_ptSquare_wSquare_WithinGap08, WeffEvent_diff_WithGap08);
+    else if (SampleIndex == 8)
+      FillBootstrap<8>(cent, ptSum_Gap08, weffEvent_WithinGap08, sum_pt_wSquare_WithinGap08, sum_ptSquare_wSquare_WithinGap08, WeffEvent_diff_WithGap08);
+    else if (SampleIndex == 9)
+      FillBootstrap<9>(cent, ptSum_Gap08, weffEvent_WithinGap08, sum_pt_wSquare_WithinGap08, sum_ptSquare_wSquare_WithinGap08, WeffEvent_diff_WithGap08);
```
[the remark this answers is not in the document]
Comment (jgrosseo) [reply PWGCF/Flow/Tasks/FlowPbPbTask.cxx:376]: You could receive the histogram pointers and put them in an array?

[the remark this answers is not in the document]
Comment (Luzhiyongg) [reply PWGCF/Flow/Tasks/FlowPbPbTask.cxx]: Hi victor, Jan,
I have implemented the changes. Please take a look.

[the remark this answers is not in the document]
Comment (jgrosseo) [reply PWGCF/Flow/Tasks/FlowPbPbTask.cxx]: > Hi victor, Jan, I have implemented the changes. Please take a look.

Sorry, I am traveling today and only on the phone.

I believe registry.add returns the pointer.

Also I was wondering if it makes sense to use the index as a histogram axis instead by adding one dimension (and using THn)

[the remark this answers is not in the document]
Comment (Luzhiyongg) [reply PWGCF/Flow/Tasks/FlowPbPbTask.cxx]: > > Hi victor, Jan, I have implemented the changes. Please take a look.
> 
> Sorry, I am traveling today and only on the phone.
> 
> I believe registry.add returns the pointer.
> 
> Also I was wondering if it makes sense to use the index as a histogram axis instead by adding one dimension (and using THn)

Hi Jan,
Yes I can simplify my code using the return pointers from registry.add(). I will do this later.
For my sight, using the index of histograms or using a TProfile2D are the same. But maybe it's due to my lack of experience and I don't know the benefits to use TProfile2D. If you have any comment on this, please tell me.

**Review by victor-gonzalez — APPROVED** [review]

File: PWGCF/Flow/Tasks/FlowPbPbTask.cxx [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+      BootstrapArray[i][0] = new TProfile(Form("hMeanPtWithinGap08_%d", i), Form("hMeanPtWithinGap08_%d", i), nMultiBins, &multiBins[0]);
+      BootstrapArray[i][1] = new TProfile(Form("c22_gap08_Weff_%d", i), Form("c22_gap08_Weff_%d", i), nMultiBins, &multiBins[0]);
+      BootstrapArray[i][2] = new TProfile(Form("c22_gap08_trackMeanPt_%d", i), Form("c22_gap08_trackMeanPt_%d", i), nMultiBins, &multiBins[0]);
+      BootstrapArray[i][3] = new TProfile(Form("PtVariance_partA_WithinGap08_%d", i), Form("PtVariance_partA_WithinGap08_%d", i), nMultiBins, &multiBins[0]);
+      BootstrapArray[i][4] = new TProfile(Form("PtVariance_partB_WithinGap08_%d", i), Form("PtVariance_partB_WithinGap08_%d", i), nMultiBins, &multiBins[0]);
+      for (int j = 0; j < 5; j++) {
+        BootstrapArray[i][j]->Sumw2();
+        fOutputList->Add(BootstrapArray[i][j]);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (Luzhiyongg) [reply PWGCF/Flow/Tasks/FlowPbPbTask.cxx]: sure, I will apply it in next PR

**Pull request merged**